### DISCUSSION TITLE
What should the deltaTime passed to a window's render event handler specifically be used for?

### DISCUSSION THREAD

**Discussion by jameswilddev** [dotnet/Silk.NET#1489] [Q&A] [answered]
When I'm setting up events for a window both the `Update` and `Render` event handlers take a delta time as an argument (which appears to be in seconds):

```
window.Update += (double deltaTime) => 
```

```
window.Render += (double deltaTime) => 
```

While I understand that the `Update` event handler should be used to advance my simulation/game state, what is intended use of the delta time passed to the `Render` event handler?

It looks like both delta times are approximately 1/60th of a second when v-synced to 60Hz, so I don't think I should be advancing my simulation/game state in `Render`, but the figure _is_ slightly different to the preceding `Update` implying that I should be doing _something_ with it and it's not just a copy of the value passed to `Update` for convenience:

```
UPDATE 0.0180284
        RENDER 0.0179364
UPDATE 0.014816
        RENDER 0.0148439
UPDATE 0.0158521
        RENDER 0.0158707
UPDATE 0.018426
        RENDER 0.0184124
UPDATE 0.0141756
        RENDER 0.0141088
UPDATE 0.0183295
        RENDER 0.0183297
UPDATE 0.0153938
        RENDER 0.0155264
UPDATE 0.0169727
        RENDER 0.0168674
UPDATE 0.0165973
        RENDER 0.0166093
UPDATE 0.0168816
        RENDER 0.0168227
```

I don't think there's a bug or anything here I'm just not quite clear on what the API is trying to communicate to me.

**Comment by Perksey** [comment, marked as answer]
The delta times for both events is the number of seconds since that particular event has been raised. Update precedes Render always.

**Comment by jameswilddev** [comment]
Ah I see thanks.  I'll just use Render's then rather than trying to combine the two.
